cxx-qt-gen/test_outputs/basic_only_signals.cpp

```diff
@@ -0,0 +1,41 @@
+#include "cxx-qt-gen/include/my_object.h"
+#include "cxx-qt-gen/src/my_object.rs.h"
+
+namespace cxx_qt::my_object {
+
+MyObject::MyObject(QObject* parent)
+  : CxxQObject(parent)
+  , m_rustObj(createRs())
+{
+  initialiseCpp(*this);
+  m_initialised = true;
+}
+
+MyObject::~MyObject() = default;
+
+void
+MyObject::invokable()
+{
+  const std::lock_guard<std::mutex> guard(m_rustObjMutex);
+  m_rustObj->invokableWrapper(*this);
+}
+
+void
+MyObject::emitReady()
+{
+  runOnGUIThread([=]() { Q_EMIT ready(); });
+}
+
+void
+MyObject::emitDataChanged(qint32 first, const QVariant& second)
+{
+  runOnGUIThread([=]() { Q_EMIT dataChanged(first, second); });
```

Comment:
Can we avoid a copy of a the QVariant here and instead move into the lambda ? As from Rust this is already a copy of QVariant that we "own".

This is the case for both opaque and non-opaque as we pass a Signal which has by-value items and opaque types are converted using to_unique_ptr.

Reply:
Should we just pass UniquePtr's for opaque types from the Rust side?